### PR TITLE
:octocat: ToDo-list: CRUD

### PR DESCRIPTION
**Pull request by mahabubx7**
### :octocat: ToDo-list: CRUD

- [x] linters-configs
- [x] GitHub workflows
- [x] Good README.md
- [x] ToDo-list: add a new task
- [x] ToDo-list: edit task
- [x] ToDo-list: remove a single task
- [x] ToDo-list: on remove => update indexes
- [x] ToDo-list: toggle task as complete
- [x] ToDo-list: clear all completed tasks
- [x] ToDo-list: reset all tasks (all clear)

### CODE REVIEW

File: src/libs/ToDo.js [hunk 1]
```diff
@@ -1,40 +1,62 @@
...
+    const indexNumber = Number(this.index);
+    this.list.push(new Task(task, indexNumber));
+    this.syncUpdates();
+  }
+
+  update(id, desc) {
+    const index = this.list.findIndex((task) => task.index === Number(id));
+    this.list[index] = { ...this.list[index], description: desc };
+    this.syncUpdates();
   }
```
Comment (Shisui6) [inline src/libs/ToDo.js:46]: Nice job storing all the add and remove changes in local storage👍. However, according to the project requirements, all your app changes should be saved in local storage. This also includes editing a task. Currently this functionality is not implemented. Please add it🙏.

**Review by Shisui6 — CHANGES_REQUESTED** [review]
Hi @mahabubx7, 

# Requires Changes 🍀
You have done a great job so far!
Your design is good 🖼️, and your code follows best practices. However, there are a few issues that you still need to work on🪲.

## Highlights
- Love your PR description and commit messages✅
- Add and remove methods work ✅
- Data saved to local storage ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

You can also consider the following:

N/A

Cheers and Happy coding!🍻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/libs/ToDo.js (hunk 1 above)
Comment (mahabubx7) [reply src/libs/ToDo.js:46]: did you read the 'syncUpdates()' method?
From line 21 to 25.

File: src/libs/ToDo.js [hunk 2]
```diff
@@ -1,40 +1,62 @@
...
+    this.list.forEach((it, index) => {
+      it.index = index + 1;
+    });
+  }
+
+  syncUpdates() {
+    this.serialize();
+    this.storage.set(this.list);
+    this.render.render();
+  }
```
Comment (mahabubx7) [inline src/libs/ToDo.js:25]: @Shisui6 It was accomplished!
look at line 23
https://github.com/mahabubx7/todo-list/blob/d207962f38518bdb9b43abb9607d2aa4d027f8de/src/libs/Storage.js#L6-L12

**Review by 1GORDON — COMMENTED** [review]
# Status: INVALID  🚨

Hi @mahabubx7 ✋

### This project will be marked as invalid and it will not affect the number of your weekly reviews.

- [x] Kindly delete the entire Live demo section from your readme file since this project is not deployed. This is a general requirement for all projects.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TadesseDev — APPROVED** [review]
### Hi @mahabubx7 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/libs/ToDo.js (hunk 1 above)
Comment (Shisui6) [reply src/libs/ToDo.js:46]: @mahabubx7, kindly double-check your work. When I edit a task, the info is not saved in local storage, as seen in the video below:

https://user-images.githubusercontent.com/37116322/228758297-1c8d43aa-2385-4db6-a1ad-9dfe45fd2871.mp4

Comment (Shisui6) [reply src/libs/ToDo.js:46]: @mahabubx7, the reviewer who approved your project did not check this. I highly encourage you to look into this and fix it so your project will be perfect

**Pull request merged**